### PR TITLE
DIV-6206: Update to case state authorisation for solicitor

### PR DESCRIPTION
**Pull request by madjava**
# JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DIV-6206

### Change description ###

Updating permissions for solicitor for DNisRefused event.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-nonprod.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[
+  {
+    "LiveFrom": "01/01/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseStateID": "DNisRefused",
+    "UserRole": "caseworker-divorce-solicitor",
+    "CRUD": "RU"
+  }
+]
```
Comment (madjava) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-nonprod.json:9]: Adding "U" for solicitor making it "RU". Changes added in the non-prod file

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json [hunk 2]
```diff
@@ -56,28 +56,28 @@
...
-    "LiveFrom": "29/07/2020",
+    "LiveFrom": "29/06/2020",
     "CaseTypeID": "DIVORCE",
     "CaseEventID": "amendPetitionForRefusalSol",
     "UserRole": "caseworker-divorce-superuser",
     "CRUD": "R"
   },
   {
-    "LiveFrom": "29/07/2020",
+    "LiveFrom": "29/06/2020",
```
Comment (madjava) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:80]: Correcting error ins liveFrom date, was meant to be 06 not 07

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json [hunk 3]
```diff
@@ -993,13 +993,6 @@
     "UserRole": "caseworker-divorce-courtadmin",
     "CRUD": "RU"
   },
-  {
```
Comment (madjava) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json:996]: Moved this to the -prod file to maintain prod config

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-prod.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[
```
Comment (madjava) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-prod.json:1]: Added this here from main file for prod

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-nonprod.json (hunk 1 above)
Comment (yarpo) [reply definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-nonprod.json:9]: You sure U is enough and we don't need C?

Comment (madjava) [reply definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-nonprod.json:9]: The citizens journey uses just 'RU' also. Went for the best least permission.

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 2 above)
Comment (Dizrupto) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:80]: Nice, 
Might also need to add the roles for caseworker-divorce-solicitor (i.e. "R").

**Comment by madjava** [comment]
Code review done, updates added and feature behind a toggle. Happy to merge this in

**Pull request merged**